### PR TITLE
migration(crons): Slugify invalid monitor slugs

### PR DESCRIPTION
**Pull request by rjo100**
Slugifys invalid monitor slugs.

Fixes https://github.com/getsentry/team-crons/issues/77

### CODE REVIEW

File: src/sentry/migrations/0506_slugify_invalid_monitors.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+def migrate_monitor_slugs(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+
+    slug_regex = re.compile("^[a-zA-Z\\d_-]+")
+
+    for monitor in RangeQuerySetWrapperWithProgressBar(Monitor.objects.filter()):
+        # Nothing to migrate if the monitor already has a valid slug
+        monitor_slug = monitor.slug
+        if slug_regex.match(monitor_slug):
+            continue
```
Comment (wedamija) [inline src/sentry/migrations/0506_slugify_invalid_monitors.py:21]: Does it make sense to just attempt to slugify and write if it changes?

File: src/sentry/migrations/0506_slugify_invalid_monitors.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+        monitor.slug = slugify(monitor_slug)[:MAX_SLUG_LENGTH].strip("-")
```
Comment (wedamija) [inline src/sentry/migrations/0506_slugify_invalid_monitors.py:23]: Better to hardcode `MAX_SLUG_LENGTH` instead of importing from external code

**Review by wedamija — APPROVED** [review]

File: src/sentry/migrations/0506_slugify_invalid_monitors.py (hunk 1 above)
Comment (rjo100) [reply src/sentry/migrations/0506_slugify_invalid_monitors.py:21]: fair

File: src/sentry/migrations/0506_slugify_invalid_monitors.py (hunk 2 above)
Comment (rjo100) [reply src/sentry/migrations/0506_slugify_invalid_monitors.py:23]: that works too

File: src/sentry/migrations/0507_slugify_invalid_monitors.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
```
Comment (markstory) [inline src/sentry/migrations/0507_slugify_invalid_monitors.py:37]: You might be close to the threshold on the number of rows that are safe to mutate during a deploy migration. 🤞

File: src/sentry/migrations/0507_slugify_invalid_monitors.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def migrate_monitor_slugs(apps, schema_editor):
+    Monitor = apps.get_model("sentry", "Monitor")
+
+    MAX_SLUG_LENGTH = 50
+
+    for monitor in RangeQuerySetWrapperWithProgressBar(Monitor.objects.filter()):
```
Comment (markstory) [inline src/sentry/migrations/0507_slugify_invalid_monitors.py:14]:
```suggestion
    for monitor in RangeQuerySetWrapperWithProgressBar(Monitor.objects.all()):
```

File: src/sentry/migrations/0507_slugify_invalid_monitors.py (hunk 3 above)
Comment (rjo100) [reply src/sentry/migrations/0507_slugify_invalid_monitors.py:37]: we're at +12k

Comment (rjo100) [reply src/sentry/migrations/0507_slugify_invalid_monitors.py:37]: and this will modify 115 rows. can link you to redash

Comment (wedamija) [reply src/sentry/migrations/0507_slugify_invalid_monitors.py:37]: <100k rows should be fine

**Pull request merged**